scripts/combine-crossreg.R

```diff
@@ -0,0 +1,237 @@
+         protocol_sponsor_linked_trn != "",
+         id != protocol_sponsor_linked_trn # or a self-reference
+  ) |>
+  mutate(is_match_protocol_sponsor_protocol_id = TRUE) |>
+  select(trn1 = id, trn2 = protocol_sponsor_linked_trn, is_match_protocol_sponsor_protocol_id) |>
+  distinct(trn1, trn2, .keep_all = TRUE)
+
+# Integrate protocol sponsor protocol matches to larger table
+trn_trn_protocols_tidy <- trn_trn_tidy |>
+  mutate(is_match_protocol_sponsor_protocol_id = NA) |>
+  rows_upsert(trn_trn_protocols, by = c("trn1", "trn2"))
+
+
+# SECOND go through results_sponsor_linked_trn
+
+# Isolate results sponsor protocol matches
+trn_trn_results <- trn_registry_data |>
+  filter(!is.na(results_sponsor_linked_trn), # filter out rows where results_sponsor_linked_trn is empty
+         results_sponsor_linked_trn != "",
+         id != results_sponsor_linked_trn # or a self-reference
+  ) |>
+  mutate(is_match_results_sponsor_protocol_id = TRUE) |>
+  select(trn1 = id, trn2 = results_sponsor_linked_trn, is_match_results_sponsor_protocol_id) |>
+  distinct(trn1, trn2, .keep_all = TRUE)
+
+# Integrate new results matches
+trn_trn_results_tidy <- trn_trn_protocols_tidy |>
+  mutate(is_match_results_sponsor_protocol_id = NA) |>
+  rows_upsert(trn_trn_results, by = c("trn1", "trn2"))
+
+
+####################################################################################################################
+# Now add matches made from the title matching algorithm. Will follow similar logic to steps above
+
+# Modify structure of title match table to make it easier to integrate with larger match table
+trn_trn_titles <- title_matches |>
+  filter(id != euctr_id) |>   # filter out self-references
+  select(trn1 = id, trn2 = euctr_id, is_title_matched = has_crossreg_eudract_tm)
+
+# Integrate title matched TRNs
+trn_trn_titles_tidy <- trn_trn_results_tidy |>
+  mutate(is_title_matched = NA) |>
+  rows_upsert(trn_trn_titles, by = c("trn1", "trn2"))
+
+####################################################################################################################
+## Finally add matches from TRNs found in publications
+
+# Unnest TRNs in pubs table
+trn_trn_pubs_long <- publications |>
+  rename(trn1 = primary_IV_id) |>
+  mutate(trn_si_long = strsplit(trns_si, ";"),
+         trn_abs_long = strsplit(trns_abs, ";"),
+         trn_ft_long = strsplit(trns_ft, ";")) |> # Split trns_si by semicolon to get individual trial IDs
+  unnest_longer(trn_si_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_si_long != trn1 | is.na(trn_si_long)) |> # filter out self-references
+  unnest_longer(trn_abs_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_abs_long != trn1 | is.na(trn_abs_long)) |> # filter out self-references
+  unnest_longer(trn_ft_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_ft_long != trn1 | is.na(trn_ft_long)) |> # filter out self-references
+  select(trn1, contains("long"))
```

Comment:
@lucfernandez: please address @delwen's https://github.com/maia-sh/intovalue-data/pull/44#discussion_r1653298457

Reply:
@maia-sh, @lucfernandez and I checked in on this last week. @lucfernandez felt it would be best to hold off on changing since it shouldn't affect the code and would require changing variable names, which could create new errors. Given limited time, I'm ok with this as long as the output would be the same (which I understood it should be?)

Reply:
I worry about introducing new changes and adding new variable names at this point. Even if this isn't as succinct or easy to understand, it works as we intend it to. Let me know your thoughts, I can change it if needed, just think it introduces unnecessary potential hurdles.

Reply:
I'm fine with leaving it for now as long as the outcome is the same as the more succinct code. @lucfernandez please add this on the docket of changes to implement down the line, thanks!